src/components/Button/Button.tsx

```diff
@@ -122,3 +138,11 @@ const StyledButton = styled.button`
     ${VARIANTS[variant]}
   `}
 `;
+
+const StyledIconButton = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 24px;
+  height: 24px;
+`;
```

Comment:
Should this be fixed value? 🤔

Reply:
I wasn't too sure if that needed a `size` like the icons because people might want the icon button rectangular and they can always set the width/height with css. That's what we were trying to prevent by setting a size to the SVG as they should be squares.

After some thought I think we can go with `buttonSize` for the icons for now, they can still overwrite it if they want; done with https://github.com/Major-Eth/yearn-web-lib/pull/23/commits/11153182bc92fcc2298ab082b60cf6ded817135a

I also noticed that I was adding the `onClick` prop to the icon rather than to the container, fixed that with the commit above as well. So now, you can have a huge container with a tiny icon and when you click outside the icon the `onClick` is still recorded as it should be. Furthermore, made the container an actual button without any style, makes sense since it's a `button` and not a `div` at the end of the day.

Reply:
Another idea could be, when using icon button we have a circle around the icon, take in the `size` prop and draw a button/clickable around the icon.

I think  Material UI does something similar(need to fact check this)

Reply:
Looking at the designs it doesn't appear that we are adding background to the icons and the icons will mostly be 24x24, so I'd say that circle is out of the scope for now. Good thinking though. What do you reckon @Major-Eth?

Reply:
Current implementation lgtm, except for the loading effect that should be removed from that I think.
Circle is out of scope for now yes.

I wonder if this should not be used directly on the Icons if a `onClick` props is used.

Additional tip: we can add something like `margin: -0.5rem, padding: 0.5rem` to add some space around the icon without moving it from it's position. Useful to increase the hitbox of the icon itself. I let you choose if it's relevant or not!

Reply:
> Current implementation lgtm, except for the loading effect that should be removed from that I think.

I thought about not having it, but then it's a button at the end of the day, so it could be in a loading state, right? Let's see a contrived example of an icon button that would render a cat or a dog depending on a network request, we do want to have it in a loading state while waiting for the response and then render either the cat or the dog. It's an optional prop anyway, but I don't have a strong opinion here. Happy to remove if you want @Major-Eth -- but think that it could be handy if it stays and it's an optional prop.

> I wonder if this should not be used directly on the Icons if a onClick props is used.

Initially I added the `onClick` on the Icon itself, but after some thought it felt wrong because it's a button at the end of the day. A normal button has text, an icon button has an icon type of thinking. It's also better to have the icon button as a styled component of a `button` rather than an `svg`, it comes with a lot of button behaviour for free, and that's what we want in a button :)